src/Compilers/Core/Portable/Operations/Expressions.cs

```diff
@@ -1,405 +1,116 @@
-
-        public IOperation Expression => _assignment;
-
-        public ITypeSymbol Type => null;
-
-        public Optional<object> ConstantValue => default(Optional<object>);
-
-        public void Accept(OperationVisitor visitor)
-        {
-            visitor.VisitExpressionStatement(this);
-        }
-
-        public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            return visitor.VisitExpressionStatement(this, argument);
-        }
-
-        private sealed class AssignmentExpression : IAssignmentExpression
-        {
-            public AssignmentExpression(IOperation target, IOperation value, SyntaxNode syntax)
-            {
-                this.Value = value;
-                this.Target = target;
-                this.Syntax = syntax;
-            }
-
-            public IOperation Target { get; }
-
-            public IOperation Value { get; }
-
-            public SyntaxNode Syntax { get; }
-
-            public ITypeSymbol Type => this.Target.Type;
-
-            public OperationKind Kind => OperationKind.AssignmentExpression;
-
-            public bool IsInvalid => Target == null || Target.IsInvalid || Value == null || Value.IsInvalid;
-
-            public Optional<object> ConstantValue => default(Optional<object>);
-
-            public void Accept(OperationVisitor visitor)
-            {
-                visitor.VisitAssignmentExpression(this);
-            }
-
-            public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-            {
-                return visitor.VisitAssignmentExpression(this, argument);
-            }
+            return (condition == null || condition.IsInvalid || ifTrue == null || ifTrue.IsInvalid || ifFalse == null || ifFalse.IsInvalid || resultType == null);
         }
     }
 
-    internal sealed class CompoundAssignment : IExpressionStatement
+    internal sealed partial class ExpressionStatement : IExpressionStatement
     {
-        private readonly CompoundAssignmentExpression _compoundAssignment;
-
-        public CompoundAssignment(IOperation target, IOperation value, BinaryOperationKind binaryOperationKind, IMethodSymbol operatorMethod, SyntaxNode syntax)
-        {
-            _compoundAssignment = new CompoundAssignmentExpression(target, value, binaryOperationKind, operatorMethod, syntax);
-            this.Syntax = syntax;
-        }
-
-        public SyntaxNode Syntax { get; }
-
-        public OperationKind Kind => OperationKind.ExpressionStatement;
-
-        public bool IsInvalid => _compoundAssignment.IsInvalid;
-
-        public IOperation Expression => _compoundAssignment;
-
-        public ITypeSymbol Type => null;
-
-        public Optional<object> ConstantValue => default(Optional<object>);
-
-        public void Accept(OperationVisitor visitor)
+        public ExpressionStatement(IOperation target, IOperation value, SyntaxNode syntax) :
```

Comment:
>ExpressionStatement [](start = 15, length = 19)

I think we shouldn't have constructors like this one and the one below. They make it very hard to understand what is going on when one is reading code that uses them.